src/pudl/etl/epacems_assets.py

```diff
@@ -44,23 +44,33 @@ def hourly_emissions_epacems(
     epacems_settings = context.resources.dataset_settings.epacems
 
     schema = Resource.from_id("hourly_emissions_epacems").to_pyarrow()
-    epacems_path = (
+    partitioned_path = (
         Path(context.op_config["pudl_output_path"]) / "hourly_emissions_epacems"
     )
-    epacems_path.mkdir(exist_ok=True)
+    partitioned_path.mkdir(exist_ok=True)
+    monolithic_path = (
+        Path(context.op_config["pudl_output_path"]) / "hourly_emissions_epacems.parquet"
+    )
+
+    with pq.ParquetWriter(
+        where=monolithic_path, schema=schema, compression="snappy", version="2.6"
+    ) as monolithic_writer:
+        for part in epacems_settings.partitions:
+            year = part["year"]
+            state = part["state"]
+            logger.info(f"Processing EPA CEMS hourly data for {year}-{state}")
+            df = pudl.extract.epacems.extract(year=year, state=state, ds=ds)
+            df = pudl.transform.epacems.transform(df, epacamd_eia, plants_entity_eia)
+            table = pa.Table.from_pandas(df, schema=schema, preserve_index=False)
+
+            # Write to one monolithic parquet file
+            monolithic_writer.write_table(table)
 
-    for part in epacems_settings.partitions:
-        year = part["year"]
-        state = part["state"]
-        logger.info(f"Processing EPA CEMS hourly data for {year}-{state}")
-        df = pudl.extract.epacems.extract(year=year, state=state, ds=ds)
-        df = pudl.transform.epacems.transform(df, epacamd_eia, plants_entity_eia)
-        with pq.ParquetWriter(
-            where=epacems_path / f"epacems-{year}-{state}.parquet",
-            schema=schema,
-            compression="snappy",
-            version="2.6",
-        ) as pqwriter:
-            pqwriter.write_table(
-                pa.Table.from_pandas(df, schema=schema, preserve_index=False)
-            )
+            # Write to a directory of partitioned parquet files
+            with pq.ParquetWriter(
```

Comment:
To maintain backward compatibility for the epacems_to_parquet command, we could add the --partition flag back. This would require adding a `partition` configuration option to this asset.

Reply:
Ah good catch.  I'm curious how much additional time doing both types of output vs. just the one takes. If it's really small, maybe it wouldn't be awful to just do both all the time? Might be confusing though.

Reply:
I've copied the implementation from dev for using the `partition` option. @bendnorman do we need to add a config option, or can we just configure it through the `dataset_settings` resource?

Reply:
It looks like to you added the  code from dev which processes EPA CEMS twice. I was thinking we could just add an asset config option that determines if we write to separate partitions:

```python
@asset(
    required_resource_keys={"datastore", "dataset_settings"},
    config_schema={
        "pudl_output_path": Field(
            EnvVar(
                env_var="PUDL_OUTPUT",
            ),
            description="Path of directory to store the database in.",
            default_value=None,
        ),
      "partition" : Field(str, default_value=True)
    },
)
def hourly_emissions_epacems(
    context, epacamd_eia: pd.DataFrame, plants_entity_eia: pd.DataFrame
) -> None:
    ....
    if context.op_config["partition"]:
    # Write to a directory of partitioned parquet files
            with pq.ParquetWriter(
                where=partitioned_path / f"epacems-{year}-{state}.parquet",
                schema=schema,
                compression="snappy",
                version="2.6",
            ) as partitioned_writer:
                partitioned_writer.write_table(table)
```

Reply:
Hmmm on second thought maybe making the partitioned version option isn't a good idea because the `epacems_io_manager` expects a directory of parquet files. This could be changed to read the monolith file instead.

Reply:
I think this should only process once still. It's just using the `partition` flag defined in `EpaCemsSettings` rather than a config option. But yeah we probably would need to change the io_manager. I don't think it's adding much time to write to both if we just want to revert back to always doing that.

Reply:
Ah sorry I read this wrong from the get go. I initially thought that the `--partition` option was to choose between partitioning and a monolithic file. Now I see it just makes the partitioned output optional. I've fixed this and made the partitioned output optional and added `partition` to the `config_schema`. I also removed `partition` from `EpaCemsSettings`, because I don't think it's necessary if we use the `config_schema`.